### ISSUE TITLE
Implement Mirror part of MeshHTTPRoute

### ISSUE THREAD

**Issue by subnetmarco** [kumahq/kuma#724]
Mirror is part of request modification in the MeshTrafficRoute MADR: https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/011-mesh-traffic-route.md#request-modification

This issue covers the implementation

### Old description

There is already an open issue https://github.com/Kong/kuma/issues/680 for TrafficMirror policy that allows to replicate traffic across multiple destinations, this is an alternative implementation that only allows one destination in the policy:

```yaml
apiVersion: kuma.io/v1alpha1
kind: TrafficMirror
mesh: default
metadata:
  namespace: default
  name: mirror-1
spec:
  sources:
    - match:
        service: backend
        version: 1
  destinations:
    - match:
        service: redis
        version: 2
  conf:
     percentage: 40
     destination:
       service: redis
       version: 3
```

And similar on Universal. Thoughts?

**Comment by jakubdyszkiewicz** [comment]
You can mirror to many destinations.

The only thing here - destinations need to be restricted to `service` only due to how we generate configuration.

**Comment by subnetmarco** [comment]
@jakubdyszkiewicz this is a non starter. The point is being able to mirror to different versions of the same service. @jakubdyszkiewicz what do we need to do to support this?

**Comment by sudeeptoroy** [comment]
This will be beneficial for universal as well.

**Comment by subnetmarco** [comment]
@jakubdyszkiewicz I believe that with 1.0 this is now unblocked since we can use `*` in the destinations, correct?

**Comment by jakubdyszkiewicz** [comment]
No, nothing changed in terms of having _multiple_ tags in destinations

**Comment by subnetmarco** [comment]
@jakubdyszkiewicz can you explain why having multiple tags in destination is more challenging?

The traffic mirroring policy is very useful across many different versions of the same service, so only specifying `kuma.io/service` would not be sufficient, as the user will most likely want to combine that with a `version` tag to the very minimum.

**Comment by lahabana** [comment]
Also see another possible design: #1031

**Comment by lahabana** [comment]
With the outcome of https://github.com/kumahq/kuma/pull/5337 Mirroring is just a part of a route. This makes things closer to GatewayAPI. I'm therefore repurposing this issue to just be the mirror implementation of the Route.

**Comment by lahabana** [comment]
Duplicate of #5796
